src/components/textarea/useResizeTextAreaEffect.ts

```diff
@@ -0,0 +1,29 @@
+import syncTextAreaHeight from './syncTextAreaHeight'
+import { useLayoutEffect } from 'react'
+
+import type { RefObject } from 'react'
+
+type UseResizeTextAreaEffectConfig = {
+  isEnabled: boolean
+  shadowTextAreaRef: RefObject<HTMLTextAreaElement>
+  textAreaRef: RefObject<HTMLTextAreaElement>
+  value?: unknown
+}
+
+export default function useResizeTextAreaEffect({
+  isEnabled,
+  shadowTextAreaRef,
+  textAreaRef,
+  value,
+}: UseResizeTextAreaEffectConfig): void {
```

Comment:
**note:** I'm using a layout effect instead of a standard effect because we're updating the DOM and we (probably) want the change to be flushed to the DOM before React's render phase completed.